### PR TITLE
Fixes #1790, changing the column selection in the "Columns to Move" field updates selection

### LINKED ISSUE #1790: Changing columns in the "Columns to Move" field should update cell selection in the table
Removing or adding columns in the modal field `Columns to Move` should also update the selection in the table. Refer to the image on the right.
Cycle 3 reference

### PR DESCRIPTION
**Pull request by AlekLefebvre**
Fixes #1790

**Technical details**
This PR propagates changes on the SelectProcessedColumns to the ColumnMode which then calls a new method SheetSelection.selectColumns which updates the selected columns.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [ ] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by seancolsen** [comment]
Thanks for this PR, @AlekLefebvre! I'm looking forward to reviewing this, but I'm going to hold off until #1750 is closed because that bug makes it harder to review/test this functionality. I've [asked](https://matrix.to/#/!UnujZDUxGuMrYdvgTU:matrix.mathesar.org/$U9LjZA85DHoQ30EmxBKcvSpMbYA8yQ63ySlVhHzFIP8?via=matrix.mathesar.org&via=matrix.org) one of our other core team members to prioritize a fix for that.

**Review by seancolsen — APPROVED** [review]
Thanks again for this PR, @AlekLefebvre! I've added some commits to clean a few things up, and I've enabled auto-merge.

- In 3abeb437f868e5d22896f4a2189c879d096ca3fd I fixed a TypeScript type that was improperly typed as a tuple when it should have been an array. Perhaps this was a simple typo, but just in case you're new to TypeScript, I'll offer a quick explanation in case this helps you with further contributions.

    Even though JS doesn't ([yet](https://github.com/tc39/proposal-record-tuple)!) have an actual tuple type, TS [_does_](https://www.typescriptlang.org/docs/handbook/2/objects.html#tuple-types). So `string[]` means _"an arbitrarily long array of strings"_, whereas `[string]` means _"an array of one and only one string"_. You can see this play out with the following code:

    ```ts
    let a: string[] = ['foo', 'bar']; // Ok

    let b: [string] = ['foo', 'bar'];
    //  ^ TypeScript error here
    // Type '[string, string]' is not assignable to type '[string]'.
    //  Source has 2 element(s) but target allows only 1
    ```

    I found this confusing when first learning TypeScript!

- In 9d93b6eb7ea83752413a78b790173c810e07a0be I gave your new method a more descriptive name and added a code comment to further clarify its behavior.

    Our UI has a feature wherein clicking on a column header will select all the cells in the column, thereby selecting the column. Given this feature, a method named `selectColumns` would appear to offer _that_ (commonly-used) functionality which selects all the rows within the supplied columns -- not the (less commonly-used) functionality of maintaining the selected rows while selecting new columns.

- In 2f841f600de2d610edc89944bbc4aaccf6081328 I cleaned up some code.

    You had forwarded the `change` event all the way from `SelectProcessedColumns`, through `ExtractColumnsModal` and `ColumnActions`, and finally to `ColumnMode` where the event was handled. That made the code in `ColumnMode.svelte` a bit opaque. It read like _"when something within the column actions UI changes, then update the column selection"_. But that's confusing because from this perspective, it's not clear _what is actually changing_ within the `ColumnActions` component. There's quite a bit of UI in there, so it could be lots of different things changing. Svelte's event system makes this even more opaque because the event handlers are loosely-typed by default, so `change.detail` actually has the type `any` which is no good.

    I moved the event handler function to reside within the `ExtractColumnsModal` component -- the component which is most pertinent to this logic. Instead of being triggered by an event, it's now triggered by a reactive statement. This is a pattern we use a lot throughout our codebase, and I think it works pretty well.

- In 5503dc259bb890fa3a9fccb3094ea1085609caff I fixed a small bug where clearing all the columns from the input caused the modal to close.

It's great to have your help! If you'd like to continue helping out, here are some more ideas:

- #1701 -- This is assigned to me, but it's pretty simple and it touches a lot of the same code as in this PR. If you like, we could even divide that ticket up into multiple parts so that you can focus only on the task of selecting the new column(s) after the modal closes. It's worth mentioning though that this ticket is pretty high priority right now and we are trying to get it closed in the next couple weeks.
- #1771
- #1892
- #1476
- #1912

**Comment by AlekLefebvre** [comment]
Thank you for the feedback, very helpful! I'll look into those other issues.

**Pull request merged**
